sdcm/cluster.py

```diff
@@ -3045,6 +3049,45 @@ def reload_config(self):
             return
         self.remoter.run(f"sudo kill -s SIGHUP {pid}")
 
+    def get_raft_status(self) -> str:
+        """ get raft state
+
+        If execute query, raft state will be return if raft is enabled,
+        otherwise result will be empty
+        Example
+
+            key                  | value
+            ----------------------+--------------------------
+            group0_upgrade_state | use_post_raft_procedures
+
+            (1 rows)
+
+
+        """
+        result = self.run_cqlsh("SELECT * FROM system.scylla_local WHERE key = 'group0_upgrade_state'",
+                                split=True, num_retry_on_failure=3)
+
+        if not result or len(result) <= 3:
+            return ""
+
+        raft_state = result[3].strip().split("|")
+        if not raft_state or len(raft_state) != 2:
+            return ""
+        return raft_state[1].strip()
+
+    def is_raft_ready(self) -> bool:
+        """check if raft running
+
+        According to https://docs.scylladb.com/branch-5.2/architecture/raft.html
+        """
+        state = self.get_raft_status()
+        return state == "use_post_raft_procedures"
+
+    def is_raft_enabled(self) -> bool:
```

Comment:
Shouldn't we cache this ? I.e. it's forbidden to disabled it, isn't it ?

Reply:
Yes it was enabled, you can't disable. But if test was running with disabled, we can enable it later. I will add cache decorator